### DISCUSSION TITLE
feature: Laravel Module

### DISCUSSION THREAD

**Discussion by spcbfr** [LazyVim/LazyVim#3476] [Extra Requests]
### Did you check the docs?

- [X] I have read all the LazyVim docs

### Is your feature request related to a problem? Please describe.

I've found laravel to be the weakest ecosystem in terms of neovim plugins. it's hard to get syntax highlighting and good LSP setup. this is especially true in the blade file format.

### Describe the solution you'd like

I module (or extra as you call it) that sets up all the laravel related packages for the user.

### Describe alternatives you've considered

Initially I tried doing everything on my own like I usually do with typescript but I found hurdles along the way. (missing autocomplete options, no blade syntax highlighting, LSP errors where there shouldn't be an error, etc..)

**Comment by kobzar** [comment]
Maybe try to use this one
https://github.com/adalessa/laravel.nvim
?

**Comment by muneebusmani** [comment]
i gotchu dude, first thing is that there is proper support for blade officially, even in vscode you would only get snippets for directives but if you reference something in Blade templates , then lsp doesnt know what you tryna do, that is the same behaviour for both neovim and vscode, phpstorm has intellisense for each and everything, and the tree sitter syntax hightlighting isnt officially supported for laravel blade files but i copied someone's setup and it has these stuff:
@if etc snippets
Blade syntax highlighting
and all other php support for the framework, i am currently trying to add my laravel setup as a lazyvim extra but couldnt find a way to do it

**Comment by MannyFay** [comment]
I have a similar problem as described [here](https://github.com/folke/lazy.nvim/discussions/1494).  
The solution for me (by now) is using Packer instead of Lazy :/

**Comment by ronald2wing** [comment]
Add Pint and blade-formatter, rustywind

**Comment by ronald2wing** [comment]
Also, enable Neotest for Laravel Sail.

**Comment by ronald2wing** [comment]
Here is a laravel extra with blade and other Laravel features https://github.com/LazyVim/LazyVim/pull/6830
